scripts/ros-generate-recipes.sh

```diff
@@ -71,41 +55,40 @@ case $ROS_DISTRO in
         ;;
 esac
 
-ROS_DISTRO_RELEASE_DATE=$3
-case $ROS_DISTRO_RELEASE_DATE in
-    none|[2-9][0-9][0-9][0-9][0-1][0-9][0-3][0-9])
-        : OK
-        ;;
-
-    *)  echo "ABORT: ROS_DISTRO_RELEASE_DATE not YYYYMMDD or 'none': '$3'"
-        exit 1
-        ;;
-esac
+if [ -n "$(git status --porcelain=v1)" ]; then
+    echo "ABORT: Uncommitted modifications detected by Git -- perhaps invoke 'git reset --hard'?"
+    exit 1
+fi
 
-if [ ! -d $4 ]; then
-    echo "ABORT: '$4' not found"
+if [ ! -f files/$ROS_DISTRO/cache.yaml ]; then
+    echo "ABORT: files/$ROS_DISTRO/cache.yaml doesn't exist, please use ros-generate-cache.sh to create it"
     exit 1
 fi
 
-if [ -n "$(git status --porcelain=v1)" ]; then
-    echo "ABORT: Uncommitted modifications detected by Git -- perhaps invoke 'git reset --hard'?"
+if [ ! -f files/$ROS_DISTRO/index-v4.yaml ]; then
+    echo "ABORT: files/$ROS_DISTRO/index-v4.yaml doesn't exist, please use ros-generate-cache.sh to create it"
     exit 1
 fi
 
-cd $4
-path_to_rosdistro=$PWD
-rev=$(git rev-list -1 $5) || exit 1
-commit_timestamp=$(git log -1 --date=iso-strict --format=%cd $rev)
-# Associate the upcoming superflore run with the commit date of <ROSDISTRO-COMMIT>.
-export SUPERFLORE_GENERATION_DATETIME=$(date +%Y%m%d%H%M%S --utc -d $commit_timestamp)
-unset commit_timestamp
-cd - > /dev/null
-
-if [ -n "$(git branch --list superflore/$SUPERFLORE_GENERATION_DATETIME)" ]; then
-    echo "ABORT: Branch 'superflore/$SUPERFLORE_GENERATION_DATETIME' already exists"
+ROS_DISTRO_RELEASE_DATE=`head -n 1 files/$ROS_DISTRO/cache.yaml | sed 's/.*cache.yaml \([^ ]*\) \([^ ]*\) \([^ ]*\)$/\1/g'`
+ROS_DISTRO_COMMIT_DATETIME=`head -n 1 files/$ROS_DISTRO/cache.yaml | sed 's/.*cache.yaml \([^ ]*\) \([^ ]*\) \([^ ]*\)$/\2/g'`
+ROS_DISTRO_COMMIT=`head -n 1 files/$ROS_DISTRO/cache.yaml | sed 's/.*cache.yaml \([^ ]*\) \([^ ]*\) \([^ ]*\)$/\3/g'`
+
+if [ -z "$ROS_DISTRO_RELEASE_DATE" -o -z "$ROS_DISTRO_COMMIT" -o -z "$ROS_DISTRO_COMMIT_DATETIME" ] ; then
+    echo "ABORT: files/$ROS_DISTRO/cache.yaml should start with # $ROS_DISTRO/cache.yaml and 3 space separated values ROS_DISTRO_RELEASE_DATE, ROS_DISTRO_COMMIT_DATETIME, ROS_DISTRO_COMMIT, please use ros-generate-cache.sh to re-create it"
     exit 1
 fi
 
+case $ROS_DISTRO_RELEASE_DATE in
```

Comment:
Why check this again? I think it can be assumed that what ros-generate-cache.sh wrote is correct and unmodified.

Reply:
When testing ros-generate-recipes.sh I've accidentally tried that on branch where cache.yaml files weren't created by latest ros-generate-cache.sh script and because the regexp which parses these variables from first line didn't match, whole first line ended in these variables (so the -z test later didn't catch that as well), so I've duplicated this part of the check to see if the  variable parsed from cache.yaml has some sane content.